### PR TITLE
SB - Creation in progress cases

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- Handling instance creation in progress when Bind request received
- Handling Bind request for failed instance
- Handling binding creation in progress when Bind request received

**Related issue(s)**

### CODE REVIEW

File: internal/broker/bind_create.go [hunk 1]
```diff
@@ -134,6 +136,20 @@ func (b *BindEndpoint) Bind(ctx context.Context, instanceID, bindingID string, d
 		expirationSeconds = parameters.ExpirationSeconds
 	}
 
+	lastOperation, err := b.operationsStorage.GetLastOperation(instance.InstanceID)
```
Comment (PK85) [inline internal/broker/bind_create.go:139]: Maybe we should consider adding a test for this new code?

File: internal/broker/bind_create.go [hunk 2]
```diff
@@ -134,6 +136,20 @@ func (b *BindEndpoint) Bind(ctx context.Context, instanceID, bindingID string, d
...
+	if lastOperation.Type == internal.OperationTypeProvision && (lastOperation.State == domain.InProgress || lastOperation.State == domain.Failed) {
+		var message string
+		if lastOperation.State == domain.InProgress {
+			message = fmt.Sprintf("instance %s creation is in progress", instanceID)
+		} else {
+			message = fmt.Sprintf("instance %s creation failed", instanceID)
+		}
+		return domain.Binding{}, apiresponses.NewFailureResponse(fmt.Errorf(message), http.StatusBadRequest, message) // Agreed with Provisioning API team to return 400
+	}
+
```
Comment (PK85) [inline internal/broker/bind_create.go:152]: Maybe we should consider adding a test for this new code?

Comment (jaroslaw-pieszka) [reply internal/broker/bind_create.go:152]: In `cmd/broker/binding_test.go` there are two test cases. One for provisioning in progress, the second for failed one.

Comment (PK85) [reply internal/broker/bind_create.go:152]: ok, I missed this test file

**Review by PK85 — APPROVED** [review]

**Review by PK85 — APPROVED** [review]

**Pull request merged**